clang/lib/CConv/DeclRewriter.cpp

```diff
@@ -0,0 +1,410 @@
+//=--DeclRewriter.cpp---------------------------------------------*- C++-*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+//===----------------------------------------------------------------------===//
+#include "clang/CConv/CCGlobalOptions.h"
+#include "clang/CConv/RewriteUtils.h"
+#include "clang/CConv/Utils.h"
+#include "llvm/Support/raw_ostream.h"
+#include "clang/CConv/DeclRewriter.h"
+#include "clang/AST/Decl.h"
+#include "clang/AST/Stmt.h"
+#include "clang/AST/ASTContext.h"
+#include "clang/Rewrite/Core/Rewriter.h"
+#include <sstream>
+#include "clang/CConv/StructInit.h"
+#include "clang/CConv/MappingVisitor.h"
+
+using namespace llvm;
+using namespace clang;
+
+unsigned int DeclRewriter::getParameterIndex(ParmVarDecl *PV,
```

Comment:
Move to `Utils.cpp` and find other code (e.g., in `ProgramInfo.cpp`) to call it (per earlier comment in `DeclRewriter.h`)

Reply:
good call. found 2 spots that duplicate the logic exactly.